### PR TITLE
Added /assignment/:id/learned

### PR DESCRIPTION
**Pull request by vietqhoang**
NOTE: Relies on #4 

Changes proposed in this pull request:

* Added PUT `/assignment/:id/learned`. Used to update state of assignments from lesson to review 

---

The credentials to view the Netlify deploy is the following:

* **username**: crabigator
* **password**: noblowfishallowed

### CODE REVIEW

**Comment by vietqhoang** [comment]
The name of the verb is still undecided. It may change.

File: source/includes/20170710/resources/_assignments.md.erb [hunk 1]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
 ### Notes
 
 The `unlocked_at`, `started_at`, `passed_at`, and `burned_at` timestamps are always in sequential order — assignments can't be started before they're unlocked, passed before they're started, etc.
+
+
+## Learned Assignment
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
## Start an Assignment
```

File: source/includes/20170710/resources/_assignments.md.erb [hunk 2]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+
+Update the assignment's state to "review". Typically used after the user has learned the associated subject in lessons.
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
Mark the assignment as started, moving the assignment from the lessons queue to the review queue. 
```

File: source/includes/20170710/resources/_assignments.md.erb [hunk 3]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+
+Assignment must be in the following valid state:
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
The assignment must be in the following valid state:
```

File: source/includes/20170710/resources/_assignments.md.erb [hunk 4]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+
+Attribute | State
+--------- | -----
+`level` | Must be less than or equal to the lower value between [User's](#user-data-structure) `level` and `max_level_granted_by_subscription`
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
`level` | Must be less than or equal to the lowest value of [User's](#user-data-structure) `level` and `max_level_granted_by_subscription`
```

Comment (oldbonsai) [reply source/includes/20170710/resources/_assignments.md.erb]: That's just a hard sentence! Maybe there's another way to phrase it, but I'm not seeing it.

File: source/includes/20170710/resources/_assignments.md.erb [hunk 5]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+Attribute | New Value
+--------- | ---------
+`available_at` | ISO8601 String timestamp
+`srs_stage_name` | `Apprentice I`
+`srs_stage` | `1`
+`started_at` | ISO8601 String timestamp
+
+> Example Request
+
+<%= partial('examples/PUT_shell', locals: { api_endpoint: 'assignments/80463006/learned' }) %>
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
<%= partial('examples/PUT_shell', locals: { api_endpoint: 'assignments/80463006/start' }) %>
```

File: source/includes/20170710/resources/_assignments.md.erb [hunk 6]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+
+<%= partial('examples/PUT_javascript', locals: { api_endpoint: 'assignments/80463006/learned' }) %>
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
<%= partial('examples/PUT_javascript', locals: { api_endpoint: 'assignments/80463006/start' }) %>
```

File: source/includes/20170710/resources/_assignments.md.erb [hunk 7]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+    "resurrected": false
+  }
+}
+```
+
+Returns the updated assignment.
+
+### HTTP Request
+
+`PUT <%= current_page.data.api_root_url %>/assignments/<id>/learned`
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]:
```suggestion
`PUT <%= current_page.data.api_root_url %>/assignments/<id>/start`
```

File: source/includes/20170710/resources/_assignments.md.erb [hunk 8]
```diff
@@ -192,3 +192,65 @@ Name | Data Type | Description
...
+    "passed": false,
+    "resurrected_at": null,
+    "resurrected": false
+  }
+}
+```
+
+Returns the updated assignment.
+
+### HTTP Request
```
Comment (oldbonsai) [inline source/includes/20170710/resources/_assignments.md.erb]: Let's scoot this up under the beginning description (line 199) and then put a header on the explanation. Per the other sections (like 'update user information') we should have an allowed parameters section for this one, too.

Comment (vietqhoang) [reply source/includes/20170710/resources/_assignments.md.erb]: It isn't clear with the snippet. Move section represented under line 254 up to line 199?

**Pull request merged**